apps/creds/tests/test_credentials_request.py

```diff
@@ -0,0 +1,112 @@
+        )
+        created_date = datetime.date(2018, 1, 1)
+        CredentialingReqest.objects.create(
+            application=test_app2)
+        CredentialingReqest.objects.filter(application=Application.objects.get(name='test_app_2')).update(created_at=created_date)
+
+        # Test App #3: credentials retrieved before, re-download/re-fetched not allowed
+        dev_user3 = self._create_user("developer_test_3", "123456")
+        test_app3 = self._create_application(
+            "test_app_3", user=dev_user3, data_access_type="THIRTEEN_MONTH"
+        )
+        CredentialingReqest.objects.create(
+            application=test_app3)
+        CredentialingReqest.objects.filter(application=Application.objects.get(name='test_app_3')).update(updated_at=created_date)
+        CredentialingReqest.objects.filter(application=Application.objects.get(name='test_app_3')).update(visits_count=1)
+
+        self.client = Client()
+
+    def test_fetch_credentials(self):
+        """
+        Tests that GET requests to /creds/{{creds_req_id}}/?action=fetch endpoint
+        retrieves and shows secret results, updates visit count, has a UUID with the correct length,
+        the right template was used to render the response, and verified template content
+        """
+        creds_req_id = CredentialingReqest.objects.get(application=Application.objects.get(name='test_app_1')).id
+        url = reverse('credentials_request', kwargs={'prod_cred_req_id': creds_req_id})
+        response = self.client.get(url, {"action": "fetch"}, follow=True)
+        self.assertEqual(len(str(creds_req_id)), 36)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'get_creds.html')
+        self.assertContains(response, 'Application: ' + 'test_app_1')
+        self.assertContains(response, 'App Credentials')
+        self.assertEqual(response.context['fetch'], 'fetch')
+        self.assertIsNotNone(response.context['client_id'])
+        self.assertIsNotNone(response.context['client_secret_plain'])
+        updated_cred_req = CredentialingReqest.objects.get(pk=creds_req_id)
+        self.assertEqual(updated_cred_req.visits_count, 1)
+
+    def test_download_credentials(self):
+        """
+        Tests that GET requests to /creds/{{creds_req_id}}/?action=download endpoint
+        retrieves secret results and downloaded JSON, updates visit count, and has a UUID with the correct length
+        """
+        creds_req_id = CredentialingReqest.objects.get(application=Application.objects.get(name='test_app_1')).id
+        url = reverse('credentials_request', kwargs={'prod_cred_req_id': creds_req_id})
+        response = self.client.get(url, {"action": "download"}, follow=True)
+        self.assertEqual(len(str(creds_req_id)), 36)
+        self.assertEqual(response.status_code, 200)
+        self.assertEquals(response["Content-Disposition"], 'attachment; filename="{}.json"'.format(creds_req_id))
```

Comment:
While assertEquals() technically works, it's a deprecated alias and looks like just a typo here considering your other calls.